### ISSUE TITLE
[BUG/ENHANCEMENT] VIP configured in a VLAN network can not be reached

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#1424]
Harvester 0.3.0

Use case: I want to configure a VIP for a guest cluster workload that is reachable from the VLAN connected to that cluster.

**Reproduce:**

1. Configure a VLAN network in Harvester
2. Configure kube-vip with a VIP cidr pool that is contained in the VLAN network range, e.g. `cidr-vms` in the example below:
```
apiVersion: v1
kind: ConfigMap
name: kubevip
namespace: kube-system
data:
  cidr-cattle-system: 192.168.6.210/32
  cidr-vms: 192.168.6.240/32 
  range-storage: 192.168.10.50-192.168.10.59
```
3. Create a guest cluster selecting the previously configured VLAN network
4. Create a workload and service loadBalancer in the guest cluster
5. Try to connect to the VIP allocated to the service

**Expected:**

The VIP is reachable from the VLAN

**Actual:**
The VIP is not reachable from the VLAN

**Additional Info:**

```
install:
  mode: create
  vip: 192.168.6.210
  vipmode: static
  networks:
    harvester-mgmt:
      interfaces:
      - name: enp1s0
      - name: enp2s0
      method: static
      ip: 192.168.6.211
      subnet_mask: 255.255.255.0
      gateway: 192.168.6.254
      #bond_options:
      #  miimon: "100"
      #  mode: active-backup
    br0-bond:
      interfaces:
      - name: enp3s0
      - name: enp4s0
      method: none
      #bond_options:
      #  miimon: "100"
      #  mode: balance-tlb
```

```
harvester5:~ # ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp1s0: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master harvester-mgmt state UP group default qlen 1000
    link/ether 52:54:00:f7:1c:73 brd ff:ff:ff:ff:ff:ff
3: enp2s0: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master harvester-mgmt state UP group default qlen 1000
    link/ether 52:54:00:85:20:b3 brd ff:ff:ff:ff:ff:ff
4: enp3s0: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master br0-bond state UP group default qlen 1000
    link/ether 52:54:00:25:49:0d brd ff:ff:ff:ff:ff:ff
5: enp4s0: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master br0-bond state UP group default qlen 1000
    link/ether 52:54:00:31:4d:17 brd ff:ff:ff:ff:ff:ff
6: harvester-mgmt: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 52:54:00:f7:1c:73 brd ff:ff:ff:ff:ff:ff
    inet 192.168.6.215/24 brd 192.168.6.255 scope global harvester-mgmt
       valid_lft forever preferred_lft forever
    inet 192.168.6.210/32 scope global harvester-mgmt
       valid_lft forever preferred_lft forever
    inet 192.168.6.240/32 scope global harvester-mgmt
       valid_lft forever preferred_lft forever
    inet 192.168.10.50/32 scope global harvester-mgmt
       valid_lft forever preferred_lft forever
    inet6 fe80::5054:ff:fef7:1c73/64 scope link 
       valid_lft forever preferred_lft forever
7: br0-bond: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue master harvester-br0 state UP group default qlen 1000
    link/ether 52:54:00:25:49:0d brd ff:ff:ff:ff:ff:ff
```

**Comment by janeczku** [comment]
Possible solution might involve running two separate kube-vip instances, one for Harvester and one for guest clusters...

**Comment by yaocw2020** [comment]
For testing
- Enable VLAN and create the VLAN network with the route. Make sure there is a DHCP server in your environment and select auto mode.
- Create a guest cluster with the harvester node driver with the harvester cloud provider. 
- Create a workload and loadBalancer service with DHCP mode in the guest cluster. 
- Try to connect to the VIP allocated to the service on the Harvester host.

**Comment by TachunLin** [comment]
Verified on `master-ee09e0c8-head` (12/7 after v1.0.0-rc1). 

### Status
Current guest cluster in harvester can use customized VLAN with auto mode to retrieve IP correctly. 
We have passed the first 3 steps of the test scenarios

1. Enable VLAN and create the VLAN network with the route. Make sure there is a DHCP server in your environment and select auto mode. 

![image](https://user-images.githubusercontent.com/29251855/145394818-151392d8-f538-497d-8c53-883aae789d72.png)


2. Create a guest cluster with the harvester node driver with the harvester cloud provider. 

![image](https://user-images.githubusercontent.com/29251855/145394864-1cbcfac8-cae5-4402-a7c7-fd28e6d01b87.png)

![image](https://user-images.githubusercontent.com/29251855/145394894-cdee434b-5d33-4352-a1ee-be185bbd5cc3.png)

3. Create a workload and loadBalancer service with DHCP mode in the guest cluster.
We can create nginx workload and loadbalancer service can get IP. 

```
hpd8s7:/home/rancher # kubectl get endpointslice
NAME                                    ADDRESSTYPE   PORTS   ENDPOINTS                                AGE
kubernetes                              IPv4          6443    10.84.44.114,10.84.44.115,10.84.44.116   21h
kubernetes-default-nginx-lb2-3e4eb608   IPv4          30905   10.84.45.115                             4h52m
kubernetes-default-ngnix-lb-616792a7    IPv4          30904   10.84.45.115                             6h42m
```

4. Try to connect to the VIP allocated to the service on the Harvester host.

### Next Action
We are trying to figure out how to correctly access the loadbalancer VIP by using curl

**Comment by TachunLin** [comment]
Verified fixed on `master-ee09e0c8-head` with rancher `v2.6.3-rc3`. Close this issue. 

### Result 
Currently can access nginx deployment and loadbalancer from harvester node. 

- [x] Pass the test scenario: Try to connect to the VIP allocated to the service on the Harvester host.


**Access rke2-cluster nodes** 
```
hpd8s7:/home/rancher # kubectl get endpointslice -o wide
NAME                                    ADDRESSTYPE   PORTS   ENDPOINTS                                AGE
kubernetes                              IPv4          6443    10.84.44.114,10.84.44.115,10.84.44.116   44h
kubernetes-default-nginx-lb2-3e4eb608   IPv4          30905   10.84.45.115                             27h
kubernetes-default-ngnix-lb-616792a7    IPv4          30904   10.84.45.115
```

```
hpd8s7:/home/rancher # curl http://10.84.44.115
<a href="https://10.84.44.115/">Found</a>.
```

**Access loadbalancer on rke2-cluster**
```
root@rke2-cluster-pool1-bde93828-k7fzs:/var/lib/rancher/rke2/bin# ./kubectl get svc
NAME         TYPE           CLUSTER-IP     EXTERNAL-IP    PORT(S)        AGE
kubernetes   ClusterIP      10.43.0.1      <none>         443/TCP        29h
nginx-lb2    LoadBalancer   10.43.87.126   10.84.45.117   80:30905/TCP   27h
ngnix-lb     LoadBalancer   10.43.61.34    10.84.45.116   80:30904/TCP   29h
```

```
hpd8s7:/home/rancher # curl http://10.84.45.117
<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
html { color-scheme: light dark; }
body { width: 35em; margin: 0 auto;
font-family: Tahoma, Verdana, Arial, sans-serif; }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
</body>
</html>
```

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-d05a6cc7-head (12/7 based on v1.0.0-rc1)
* Rancher version: v2.6.3-rc3

### Verify Steps

1. Enable virtual network with `harvester-mgmt` in harvester
2. Create VLAN 806 with id `806` and set to default `auto` mode
3. Import harvester to rancher
4. Create cloud credential
5. Provision a rke2 cluster to harvester 
![image](https://user-images.githubusercontent.com/29251855/145564732-0a3cee15-a264-407f-800a-df2e7c649846.png)

![image](https://user-images.githubusercontent.com/29251855/145564961-c921f341-2c88-44cc-9c5e-08789e594552.png)

6. Deploy a `nginx` server workload 
![image](https://user-images.githubusercontent.com/29251855/145565066-c9701c25-c0c6-4077-9718-80279d1d9765.png)

![image](https://user-images.githubusercontent.com/29251855/145565122-41f2fa46-b6de-4a54-bd4e-afe2b9cc6bef.png)

7. Open Service Discover -> Services 
8. Create a Load Balancer service
![image](https://user-images.githubusercontent.com/29251855/145565240-40cd93d7-aadb-4aed-9b99-e0878f014071.png)

9. Provide service port 
![image](https://user-images.githubusercontent.com/29251855/145565319-64191c22-6961-4332-ab13-a6373b4c3814.png)

10.   Provide health check port
![image](https://user-images.githubusercontent.com/29251855/145565383-934f3170-c141-4ebc-bffa-3afb7f32282e.png)

11.  Edit yaml on nginx workload , copy the lable key value pair
![image](https://user-images.githubusercontent.com/29251855/145565663-c768bedc-9286-408c-9a6b-6fc11056a225.png)

12.  Open `Selectors` tab in loadbalncer 
13.  Paste the key value pair from nginx 
![image](https://user-images.githubusercontent.com/29251855/145565921-199188c8-0eb7-4ae4-9630-4219eea8cecf.png)

![image](https://user-images.githubusercontent.com/29251855/145565961-0cd66b49-7c20-4036-a4c2-d5f34dfe3a18.png)

14. Open Cluster -> Nodes
15. Access the RKE2 nodes
![image](https://user-images.githubusercontent.com/29251855/145566203-38f44c55-ee39-4095-a8fb-946efdcb3176.png)

16. Click the `Download SSH key`
17. unzip the ssh key 
18. Run `ssh -i <id_rsa> to login rke2 cluster 
19. Access RKE2 cluster 
https://docs.rancher.cn/docs/rke2/cluster_access/_index/
20. Run `./kubectl get svc` 
21. ssh to harvester node 1
22. Run `kubectl get endpointslice -o wide` 
23. Run `curl http://{nigix endpoint ip}
24. Run `curl http://{loadbalcner external ip}
